Improve user experience of `idris-switch-to-repl` command

keram opened the pull request:
After testing these changes locally for some time, stabilised the changes to:

- load current visiting Idris buffer file if `idris-switch-to-repl` is executed with prefix
  Example: `C-- C-c C-z`
  This aligns with similar behaviour of other languages in Emacs particularly clojure (cider).
  The loading is performed synchronuosly and optimistically, with errors being ignored.

- If active region paste region into the REPL

  This makes experimentation with code simpler.

gallais commented on idris-repl.el:
```diff
@@ -186,9 +186,16 @@ If ALWAYS-INSERT is non-nil, always insert a prompt at the end of the buffer."
   (interactive)
+  (if (and buffer-file-name
+           ;; in Emacs 29.1 > we can use string-equal-ignore-case
+           (string= "idr" (downcase (file-name-extension buffer-file-name))))
```
Is it not the case that idris files can have other extensions if we are in literate mode?

keram replied:
Good point! I have updated the implementation to check also for `lidr` files although I don't have any good lidr examples to test it with.

keram committed:
Suppress Idris errors when loading a file in `idris-switch-to-repl`

This improves the user workflow.

When writing Idris code, it is often useful to switch back and forth
to the REPL (e.g., to test functions, look up docs, or check types),
even when the code is in a broken state.
However, when an error occurs, switching to the REPL is interrupted,
leaving the point in the original buffer and forcing the user
to dismiss the Idris notes.

keram committed:
Ensure Idris is running and current Idris is file loaded when using `idris-switch-to-repl`

Why:
Allows to jump to the repl without having to manually start repl
using a shortcut

keram commented on idris-repl.el:
```diff
@@ -186,9 +186,12 @@ If ALWAYS-INSERT is non-nil, always insert a prompt at the end of the buffer."
-  "Select the output buffer and scroll to bottom."
-  (interactive)
+  "Load the current Idris file buffer and jump to the Idris REPL."
+  (interactive nil idris-mode)
```
Specifying `idris mode` should exclude this command from M-x list in non idris-mode based buffers. Depends on user setting as per documentation:

```
If MODES is present, it should be one or more mode names (symbols)
for which this command is applicable.  This is so that ‘M-x TAB’
will be able to exclude this command from the list of completion
candidates if the current buffer’s mode doesn’t match the list.
Which commands are excluded from the list of completion
candidates based on this information is controlled by the value
of ‘read-extended-command-predicate’, which see.
```

keram committed:
Load current file only if command executed with prefix.

2. This aligns with similar behaviour of other languages in emacs
  particularly clojure (cider).

keram committed:
Experimental paste selected region into Idris repl

Requires `idris-x-switch-to-repl-copy-region`
to be enabled (Enabled by default).

jfdm commented:
Is this ready to go?

keram committed:
Allow idris-load-file-sync to ignore errors

Why:
When jumping to a repl from non-loaded file
we want to do optimistic load.
Ergo even if Idris finds errors in the file the jump
to the repl should succeed.

keram commented:
> Is this ready to go?

Yes, it is. 🤞🏻 
 I have rebased the branch against `main` to resolve conflicts.
Also to be bit defensive moved "copy region" to repl behind a feature flag.

In theory after all changes `idris-repl` command could be an alias for `idris-switch-to-repl` 
but keeping them separate for now.
